middleware/expvar/expvar.go

```diff
@@ -0,0 +1,46 @@
+package expvar
+
+import (
+	"expvar"
+	"fmt"
+	"net/http"
+
+	"github.com/mholt/caddy/middleware"
+)
+
+// ExpVar is a simple struct to hold expvar's configuration
+type ExpVar struct {
+	Next     middleware.Handler
+	Resource Resource
+}
+
+// ServeHTTP handles requests to expvar's configured entry point with
+// expvar, or passes all other requests up the chain.
+func (e ExpVar) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error) {
+	if middleware.Path(r.URL.Path).Matches(string(e.Resource)) {
+		expvarHandler(w, r)
+		return http.StatusOK, nil
```

Comment:
This should be a `return 0, nil`, indicating that your handler wrote to the response. See https://godoc.org/github.com/mholt/caddy/middleware#Handler

> If ServeHTTP writes to the response body, it should return a status code of 0. This signals to other handlers above it that the response body is already written, and that they should not write to it also.